Added Health Check Endpoint

TheCoderAdi opened the pull request:
## Description
This PR introduces a /health endpoint in the FastAPI application to monitor the MongoDB connection status and response time. This helps ensure the application is operating as expected and provides insight into the database's health.

## Related Issue
Fixes #12 

## Motivation and Context
- Check database connectivity
- Report connection status
- Add response time metrics

## How Has This Been Tested?
Go to the /health route
- Healthy Scenario: Verified the endpoint responds with status: healthy when MongoDB is connected.
- Unhealthy Scenario: Simulated MongoDB connection issues to verify the endpoint reports an error with status: unhealthy.


## Screenshots (if appropriate):
https://github.com/user-attachments/assets/62399c19-8ac4-4bba-84f8-47c56b8275d8



## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist:
- [x] My code follows the code style of this project
- [x] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings

Varsha-1605 commented:
@TheCoderAdi  where is this output seen?? can you show it in the form of a notification saying something like the database's health is good and if not that must be shown. (healthy/unhealthy)

TheCoderAdi commented:
> @TheCoderAdi where is this output seen?? can you show it in the form of a notification saying something like the database's health is good and if not that must be shown. (healthy/unhealthy)

What I have done?
I have added a route /health route where if you will go it will give you the response.


Okay so when a user goes to /health route then in the home page it will show a popup of the status
right?

Varsha-1605 commented:
can you show me both the cases when connected and disconnected?

TheCoderAdi commented:
Sure 
this is the unhealthy status ,

Steps to recreate:
I have deleted the previously cached MongoDB URL
using the command - Remove-Item Env:MONGODB_URL

Then I have added wrong MongoDB URL in the .env
Then I visited the page /health

this is the output :

![image](https://github.com/user-attachments/assets/a09b2652-fa63-4e98-812b-cebc0dfcbd7c)
